storage: extract post-commit portion of split

tbg opened the pull request:
This is the first step towards formalizing the split trigger as a side effect
in the sense of #6286. With proposed-evaluated KV, we will have to attach
a small amount of metadata to the WriteBatch containing the SplitTrigger based
from which we have to deduce the necessary actions.

I have more WIP which contains plumbing-type work, but have decided to take
a step back and bring these changes in in small increments as a plethora of
roadblocks is to be expected and #6144 may influence (and/or be influenced by)
this work.

Code movement only (and changing a few words around the boundary of the move) in one of the commits, the other one has the actual changes without spurious diff.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/7915)

<!-- Reviewable:end -->

tbg committed:
storage: prepare parts of split trigger for extraction

tamird commented:
Reviewed 1 of 1 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[storage/replica_command.go, line 2305 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7915#-KNEzXwi5NxblQzDU2IZ:-KNEzXwi5NxblQzDU2I_:166874) ([raw file](https://github.com/cockroachdb/cockroach/blob/50b436817eac3b90f1506fc7b5bb03854228bdcc/storage/replica_command.go#L2305)):*
> ```Go
> 
> 	// Update store stats with difference in stats before and after split.
> 	// TODO(tschottdorf): at this point
> ```

?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7915)*
<!-- Sent from Reviewable.io -->

bdarnell commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[storage/replica_command.go, line 2644 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7915#-KNPfBLi_JGGKyOugzh5:-KNPfBLi_JGGKyOugzh6:465947029) ([raw file](https://github.com/cockroachdb/cockroach/blob/50b436817eac3b90f1506fc7b5bb03854228bdcc/storage/replica_command.go#L2644)):*
> ```Go
> 	// Note: you must not use the trace inside of this defer since it may
> 	// run after the trace has already completed.
> 	// TODO(tschottdorf): the above doesn't sound right.
> ```

I think that comment has drifted; it belongs on the RunAsyncTask call inside the trigger. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7915#-:-KNPfh6zmb3-XqMCmdNI:1417088721)*
<!-- Sent from Reviewable.io -->

tbg committed:
storage: extract post-commit portion of split

This is the first step towards formalizing the split trigger as a side effect
in the sense of #6286. With proposed-evaluated KV, we will have to attach
a small amount of metadata to the WriteBatch containing the SplitTrigger based
from which we have to deduce the necessary actions.

I have more WIP which contains plumbing-type work, but have decided to take
a step back and bring these changes in in small increments as a plethora of
roadblocks is to be expected and #6144 may influence (and/or be influenced by)
this work.

Code movement only (and changing a few words around the boundary of the move)
in this commit.

tschottdorf commented:
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[storage/replica_command.go, line 2305 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7915#-KNEzXwi5NxblQzDU2IZ:-KNXr8FlXABMKqboBs9f:1740281843) ([raw file](https://github.com/cockroachdb/cockroach/blob/50b436817eac3b90f1506fc7b5bb03854228bdcc/storage/replica_command.go#L2305)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> ?
</details>
Removed.

---

*[storage/replica_command.go, line 2644 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7915#-KNPfBLi_JGGKyOugzh5:-KNXr1a4OlvUJOkfX0Y3:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/50b436817eac3b90f1506fc7b5bb03854228bdcc/storage/replica_command.go#L2644)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary>
> I think that comment has drifted; it belongs on the RunAsyncTask call inside the trigger. 
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7915)*
<!-- Sent from Reviewable.io -->
